### PR TITLE
Move cookie list override logic to account for uninitialized preferences

### PR DESCRIPTION
**Pull request by antonok-edm**
The previous implementation only iterated through lists that existed in the preferences already, so in practice it would only override lists that had `enabled: false` - i.e. lists that had been enabled and disabled again before the "touched" preference was introduced.

This fix clones the preferences dictionary and adds the cookie list with `enabled: true` to the cloned dictionary if the override should occur.

Resolves https://github.com/brave/brave-browser/issues/20308

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Same as https://github.com/brave/brave-core/pull/11621

### CODE REVIEW

File: components/brave_shields/browser/ad_block_regional_service_manager.cc [hunk 1]
```diff
@@ -70,18 +70,24 @@ void AdBlockRegionalServiceManager::StartRegionalServices() {
   base::AutoLock lock(regional_services_lock_);
   const base::DictionaryValue* regional_filters_dict =
       local_state->GetDictionary(prefs::kAdBlockRegionalFilters);
-  for (base::DictionaryValue::Iterator it(*regional_filters_dict);
-       !it.IsAtEnd(); it.Advance()) {
-    const std::string uuid = it.key();
+
+  std::unique_ptr<base::Value> regional_filters_dict_with_cookielist =
+      std::make_unique<base::Value>(regional_filters_dict->Clone());
```
Comment (emerick) [inline components/brave_shields/browser/ad_block_regional_service_manager.cc:75]: Do we need to use a `std::unique_ptr` here? Can we just do this without a pointer?

**Review by emerick — APPROVED** [review]
LGTM, just had one minor comment - up to you if you make that change, though.

File: components/brave_shields/browser/ad_block_regional_service_manager.cc (hunk 1 above)
Comment (antonok-edm) [reply components/brave_shields/browser/ad_block_regional_service_manager.cc]: Yep, that's a good call - I was looking at some older code which was still using `base::DictionaryValue`. Fixed now.

**Comment by antonok-edm** [comment]
Android CI failure is unrelated, merging.

**Comment by stephendonner** [comment]
Verified `PASSED` using

Brave | 1.36.14 Chromium: 97.0.4692.71 (Official Build) nightly (64-bit)
-- | --
Revision | adefa7837d02a07a604c1e6eff0b3a09422ab88d-refs/branch-heads/4692@{#1247}
OS | Windows 10 Version 20H2 (Build 19042.1415)

Followed the testplan from https://github.com/brave/brave-core/pull/11621.

Steps:

1. Visit https://digikey.com and confirm that a cookie notice is shown. It will appear either as a banner at the top, or as a popup along the bottom depending on window size.
2. Visit brave://flags and enable the #brave-adblock-cookie-list-default flag, then restart the browser.
3. Visit https://digikey.com and confirm that no cookie notice is shown.
4. Visit brave://adblock and confirm that the entry for Easylist-Cookie List - Filter Obtrusive Cookie Notices under Additional filters exists and is checked.
5. Return to brave://flags and set #brave-adblock-cookie-list-default to Disabled, and restart the browser.
6. Visit https://digikey.com and confirm that the cookie notice is shown again.
7. Visit brave://adblock and confirm that the entry for Easylist-Cookie List - Filter Obtrusive Cookie Notices is no longer checked.
8. Manually toggle the Easylist-Cookie List - Filter Obtrusive Cookie Notices entry to be enabled, then repeat steps 2-5.
9. Visit brave://adblock and confirm that the entry for Easylist-Cookie List - Filter Obtrusive Cookie Notices is checked.
10. Manually toggle the Easylist-Cookie List - Filter Obtrusive Cookie Notices entry to be disabled.
11. Visit brave://flags and enable the #brave-adblock-cookie-list-default flag, then restart the browser.
12. Visit brave://adblock and confirm that the entry for Easylist-Cookie List - Filter Obtrusive Cookie Notices is unchecked.
13. Visit https://digikey.com and confirm that the cookie notice is shown.

step1 | step 2 | step 3 | step 4 | step 5 | step 6 | step 7
-------|--------|--------|--------|--------|--------|--------
PR-11730-2 | PR-11730-1 | PR-11730-3 | PR-11730-4 | PR-11730-5 | PR-11730-6 | PR-11730-7


step 8a | step 8b | step 8c | step 9 | step 10 | step 11 | step 12 | step 13
--------|---------|---------|----------|---------|---------|----------|---------
 | PR-11730-8 |  PR-11730-9 | PR-11730-10 | PR-11730-11 | PR-11730-12 | PR-11730-13 | PR-11730-14 | PR-11730-15

**Pull request merged**